mono/metadata/file-mmap-posix.c

```diff
@@ -226,7 +226,7 @@ XXX implement options
 static void*
 open_file_map (MonoString *path, int input_fd, int mode, gint64 *capacity, int access, int options, int *error)
 {
-	struct stat buf;
+	struct stat buf = { 0 };
```

Comment:
We should not use this idiom to initialize the structure.

Instead we should check if the structure is actually initialized.   So if we fail, we should either bail out, or record the fact that stat failed on a separate flag.